assets/js/components/higherorder/withdata.js

```diff
@@ -62,7 +62,12 @@ import { __ } from '@wordpress/i18n';
  *                                                 if the data is "zero".
  * @param {function}        getDataError           A callback function that is passed the resolved data and returns the
  *                                                 error message.
-
+ *
+ * @return {React.Component} Component  	Returns React.Components based on data and state.
```

Comment:
Not sure if this should be `React.Component` or the one that comes from `@wordpress/element`. I defer to @tofumatt (see other comment below).

Reply:
I would say that using `WPElement` is more appropriate. It's a pass-through, sure, but referencing it is more accurate/future-proof.

I'm not sure how to declare it here. Probably just `Component` as that's how it's imported so will make sense. I'm not sure how JSDoc handles types that aren't in the file—I guess it doesn't care. Labelling this `Component` kinda sucks, but is sort of more accurate than `React.Component`. But for functional components that's gonna be weird 😢